internal/backend/sftp/sftp.go

```diff
@@ -258,23 +258,40 @@ func (r *SFTP) Save(ctx context.Context, h restic.Handle, rd restic.RewindReader
 	}
 
 	filename := r.Filename(h)
+	dirname := r.Dirname(h)
 
-	// create new file
-	f, err := r.c.OpenFile(filename, os.O_CREATE|os.O_EXCL|os.O_WRONLY)
+	// Ensure that the containing directory exists.
+	if err := r.c.MkdirAll(dirname); err != nil {
```

Comment:
When backing up to a remote SFTP server, this adds a full round-trip just to create a directory which usually already exists. I'd like to keep the old optimistic behavior of just opening the file and then creating the directory if that didn't work.

Reply:
This was added because the directory has to exist for StatVFS. But if we check after the fact, it's not needed.